src/tests/Microsoft.PowerFx.Interpreter.Tests/LanguageServiceProtocol/LanguageServerTests.cs

```diff
@@ -1376,6 +1378,373 @@ public void CustomCheckResult()
             Assert.Contains("The type of this expression does not match the expected type 'Text'. Found type 'Decimal'.", diags.First().Message);
         }
 
+        [Fact]
+        public void TestCorrectFullSemanticTokensAreReturned()
+        {
+            // Arrange
+            var expression = "Max(1, 2, 3)";
+            var semanticTokenParams = new SemanticTokensParams
+            {
+                TextDocument = new TextDocumentIdentifier { Uri = GetUri($"expression={expression}") }
+            };
+            var payload = GetFullDocumentSemanticTokensRequestPayload(semanticTokenParams);
+
+            // Act
+            _testServer.OnDataReceived(payload.payload);
+
+            // Assert
+            var response = AssertAndGetSemanticTokensResponse(_sendToClientData?.FirstOrDefault(), payload.id);
+            Assert.NotEmpty(response.Data);
+            var decodedTokens = SemanticTokensRelatedTestsHelper.DecodeEncodedSemanticTokensPartially(response, expression);
+            Assert.Single(decodedTokens.Where(tok => tok.TokenType == TokenType.Function));
+            Assert.Equal(3, decodedTokens.Where(tok => tok.TokenType == TokenType.NumLit || tok.TokenType == TokenType.DecLit).Count());
+        }
+
+        [Theory]
+        [InlineData("Create", TokenType.Function, TokenType.BoolLit)]
+        [InlineData(null)]
+        [InlineData("")]
+        [InlineData("[2, 3")]
+        [InlineData("Create", TokenType.StrInterpStart, TokenType.BinaryOp, TokenType.NumLit, TokenType.DecLit, TokenType.Control)]
+        [InlineData("[]")]
+        [InlineData("1,2]")]
+        [InlineData("[98]")]
+        [InlineData("Create", (TokenType)78)]
```

Comment:
>(TokenType)78 [](http://example.com/codeflow?start=30&length=13)

where is 78 coming from?
ie, if the test starts returning 79, how do we know if that's a  product bug or a test bug?

Reply:
I wanted to test a scenario where client would pass in a token type that it wants to skip but the token type is not a valid one (a.k.a out of range). In this case, no tokens should be skipped as 78 is not one of the values of any token type (the max is 26). I can change this to TokenType.Lim which is also considered as not a valid token type. I compute a list of all possible token types in the range within a formula without skipping any token type and then i re compute the tokens in the same range this time with skipping the token types and then asserting against the list of possible token types in the range before skipping with the list of possible token types after skipping. For this case both list should be same as 78 doesn't represent a valid token type and therefore no tokens should be skipped.

Reply:
felt like this test was a bit magic so made it a less magic by manually setting all possible expected token types in the range i am using for the test. but the expectation remains the same (78 or TokenType.Lim) are not considered valid token types and since there would be no token types with these two values, none of the expected token types should be skipped